File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def bt(n, numbers, idx, current, target):    
    global answer
    
    if idx == n:
        if current == target:
            answer += 1
        
        return
        
    bt(n, numbers, idx + 1, current + numbers[idx], target)
    bt(n, numbers, idx + 1, current - numbers[idx], target)
    

def solution(numbers, target):
    global answer
    
    answer = 0
    
    bt(len(numbers), numbers, 0, 0, target)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.kt
class Solution {
    fun dfs(numbers: IntArray, target: Int, idx: Int, total: Int): Int {
        if (idx == numbers.size) {
            if (total == target) return 1 else return 0
        }
        
        return dfs(numbers, target, idx + 1, total + numbers[idx]) + dfs(numbers, target, idx + 1, total - numbers[idx])
    }
    
    
    fun solution(numbers: IntArray, target: Int): Int {
        return dfs(numbers, target, 0, 0)
    }
}